dietpi/dietpi-software

```diff
@@ -4166,6 +4173,55 @@ WantedBy=multi-user.target
 _EOF_
 		fi
 
+		if To_Install 141 adsb-feeder # ADSB Feeder
+		then
+			# create the userdata directory and clone the adsb-feeder repo into /tmp
+			G_EXEC mkdir -p /mnt/dietpi_userdata/adsb-feeder
+			G_EXEC_OUTPUT=1 G_EXEC git clone -b dietpi 'https://github.com/dirkhh/adsb-feeder-image.git' /tmp/adsb-feeder
+
+			# remove the service that isn't needed for an app install on DietPi and install the rest
+			G_EXEC cd /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root
+			G_EXEC rm -rf ./usr/lib/systemd/system/adsb-bootstrap.service
+			G_EXEC cp -a ./usr/lib/systemd/system/* /etc/systemd/system/
+
+			# determine the version
+			ADSB_FEEDER_DATE_COMPONENT=$(git log -20 --date=format:%y%m%d --format="%ad" | uniq -c | head -1 | awk '{ print $2"."$1 }')
+			ADSB_FEEDER_TAG_COMPONENT=$(git describe --match "v[0-9]*" | cut -d- -f1)
+			ADSB_FEEDER_VERSION="${ADSB_FEEDER_TAG_COMPONENT}(dietpi)-${ADSB_FEEDER_DATE_COMPONENT}"
+
+			# create the target directory for the app and populated with the code from the git checkout
+			G_EXEC mkdir -p /opt
+			G_EXEC cp -a /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root/opt/adsb /opt
+			G_EXEC cd /opt/adsb
+
+			# remove the git clone of the repo we installed from
+			G_EXEC rm -rf /tmp/adsb-feeder
+
+			# create a symlink so the config files reside where they should be in /mnt/dietpi_userdata/adsb-feeder
+			G_EXEC mkdir -p /mnt/dietpi_userdata/adsb-feeder/config
+			G_EXEC ln -s /mnt/dietpi_userdata/adsb-feeder/config .
+
+			# set the 'image name' and version that are shown in the footer of the Web UI
+			echo "ADSB Feeder app running on DietPi" > feeder-image.name
+			echo "$ADSB_FEEDER_VERSION" > adsb.im.version
+
+			# get the Python module
+			G_AGI python3-flask
```

Comment:
Not sure whether it is a benefit, just the idea of installing an up-to-date flask via `pip`:
```suggestion
			G_EXEC_OUTPUT=1 G_EXEC pip3 install -U flask
```

Reply:
It's an interesting question, though. I had one case with old versions of flask included by an OS actually broke my build, but anything that's relatively current I prefer the OS included one as that gets updated and receives patches.

Are all your builds on bookworm at this point?

Reply:
New Bullseye images are also still provided and Buster systems are still supported. Hence oldest is flask 1.0.2.

Reply:
ok. that sucks.
I need to work around that then.

Reply:
We could also disable/hide the software option in Buster and Bullseye (flask 1.1.2), or use `pip` only on those, in case with a fixed/compatible version, like `^2.2.2` to match the one on Bookworm.

If any, I guess flack 1.x vs 2.x is causing issues?

Reply:
I'm trying to please.
Actually, I'm trying to get this ready to be merged before I'm heading on a trip tomorrow.